tests/grid/tria_accessor_neighbors.cc

```diff
@@ -0,0 +1,388 @@
+    RefinementCase<2>::isotropic_refinement,
+    RefinementCase<2>::cut_axis(1),
+    RefinementCase<2>::cut_axis(1),
+    RefinementCase<2>::isotropic_refinement,
+    RefinementCase<2>::isotropic_refinement,
+    RefinementCase<2>::isotropic_refinement,
+    RefinementCase<2>::isotropic_refinement,
+    RefinementCase<2>::cut_axis(0),
+    RefinementCase<2>::isotropic_refinement,
+    RefinementCase<2>::cut_axis(0),
+    RefinementCase<2>::cut_axis(1),
+    RefinementCase<2>::isotropic_refinement,
+    RefinementCase<2>::cut_axis(1),
+    RefinementCase<2>::cut_axis(1),
+    RefinementCase<2>::isotropic_refinement,
+    RefinementCase<2>::isotropic_refinement,
+    RefinementCase<2>::cut_axis(1),
+    RefinementCase<2>::isotropic_refinement,
+    RefinementCase<2>::isotropic_refinement,
+    RefinementCase<2>::isotropic_refinement,
+    RefinementCase<2>::isotropic_refinement,
+    RefinementCase<2>::isotropic_refinement,
+    RefinementCase<2>::isotropic_refinement,
+    RefinementCase<2>::isotropic_refinement,
+    RefinementCase<2>::isotropic_refinement,
+    RefinementCase<2>::isotropic_refinement,
+    RefinementCase<2>::cut_axis(0),
+    RefinementCase<2>::isotropic_refinement,
+    RefinementCase<2>::isotropic_refinement,
+    RefinementCase<2>::isotropic_refinement,
+    RefinementCase<2>::cut_axis(0),
+    RefinementCase<2>::isotropic_refinement,
+    RefinementCase<2>::isotropic_refinement,
+    RefinementCase<2>::isotropic_refinement};
+  return out;
+}
+
+std::vector<unsigned int>
+get_cell_numbers()
+{
+  std::vector<unsigned int> out = {
+    2,   5,   5,   12,  2,   16,  2,   4,   18,  15,  8,   21,  13,  19,  11,
+    15,  10,  9,   56,  50,  63,  23,  50,  89,  82,  96,  38,  105, 4,   73,
+    38,  81,  89,  29,  143, 65,  166, 135, 12,  130, 46,  213, 132, 71,  106,
+    102, 0,   226, 258, 175, 278, 196, 57,  272, 288, 260, 218, 33,  337, 86,
+    255, 267, 163, 179, 223, 175, 291, 347, 260, 272, 410, 369, 4,   90,  170,
+    417, 361, 275, 413, 287, 218, 362, 375, 476, 69,  24,  138, 195, 149, 32,
+    219, 252, 112, 127, 583, 415, 583, 522, 288, 125};
+  return out;
+}
+
+
+void
+test()
+{
+  // Get a grid in 2D
+  Triangulation<2> triangulation;
+  GridGenerator::hyper_cube(triangulation);
+  triangulation.refine_global();
+
+  const std::vector<RefinementCase<2>> refinements  = get_refinement_cases();
+  const std::vector<unsigned int>      cell_numbers = get_cell_numbers();
+
+  // Generate some random refinement
+  // The idea is as follows
+  // ref_type is a random number between 0 and 2.
+  //  -> If ref_type is 0, 1 we refine that axis
+  //  -> If ref_type is 2 we use isotropic refinement
+  // cell_number refers to the cell we want to refine
+  // obtained by advancing triangulation.begin_active()
+  // cell_number times
+  for (unsigned int r = 0; r < 100; r++)
+    {
+      auto cell = triangulation.begin_active();
+      for (unsigned int i = 0; i < cell_numbers[r]; i++, cell++)
+        ;
```

Comment:
Did you mean to have an empty `for`-loop body?

Reply:
Yup. I increment the cell iterator inside the `(...)`. Could be done in the body as well.

Reply:
You shoudl be able to write this as
```suggestion
       std::advance(cell, cell_numbers[r]);
```
for clarity (not tested).

Reply:
Ah yeah, that's better. I tested it locally.